spec/services/search/buffer_suggestions_builder_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "rails_helper"
+include DistanceHelper
+
+RSpec.describe Search::BufferSuggestionsBuilder do
+  let(:subject) { described_class.new(location, search_params) }
+
+  describe "#get_buffers_suggestions" do
+    let(:search_params) do
+      {
+        hits_per_page: 10,
+      }
+    end
+    let(:location) { "Tower Hamlets" }
+    let(:buffer_coordinates) { [5.004562496029994090, 56.50833566307333, 5.005710530794815389, 56.5051084208278] }
+    let(:buffer_hash) do
+      {
+        "5" => buffer_coordinates,
+        "10" => buffer_coordinates,
+        "15" => buffer_coordinates,
+        "20" => buffer_coordinates,
+        "25" => buffer_coordinates,
+      }
+    end
+
+    before do
+      allow(LocationPolygon).to receive_message_chain(:with_name, :buffers).and_return(buffer_hash)
+    end
+
+    context "when there is one vacancy within all of the buffer polygons" do
```

Comment:
Ambiguous
"when there is only one vacancy in any of the buffer polygons"
it "suggests only the relevant buffer"